app/market/item/[id]/page.jsx

```diff
@@ -1,43 +1,147 @@
-"use client";
-
-import { useEffect, useState } from "react";
-import { useRouter } from "next/navigation";
-import { Box, Button } from "@mui/material";
-import CardComponent from "@/components/CardComponent";
-
-/**
- * @param {params} Object
- 
-*/
-export default function Page({ params }) {
-  const [data, setData] = useState(null);
+'use client'
+import * as React from "react";
+import { Typography, Button, Card, CardMedia, ThemeProvider, Breadcrumbs, Link } from "@mui/material";
+import { useTheme } from "@mui/material/styles";
+import { useRouter } from 'next/navigation'; // Importing useRouter from next/navigation
+import ShoppingCartIcon from '@mui/icons-material/ShoppingCart';
+import FavoriteIcon from '@mui/icons-material/Favorite';
+
+
+
+const IndividualCardPage = ({ params }) => {
+  const [cardDetails, setCardDetails] = React.useState(null);
   const router = useRouter();
+  console.log(params)
+  const theme = useTheme();
   const id = params.id;
-  useEffect(() => {
-    //fetch data need to move to file in utils
-    const fetchCard = async () => {
-      const response = await fetch(`/api/cards/${id}`);
-      if (!response.ok) {
-        throw new Error(response.statusText);
+  console.log(id)
+
+
+
+
+  React.useEffect(() => {
+    const fetchCardDetails = async () => {
+      try {
+        const response = await fetch(`/api/cards/${id}`);
+        if (!response.ok) {
+          throw new Error('Failed to fetch card details');
+        }
+        const data = await response.json();
+        console.log(data);
+        setCardDetails(data.data);
+      } catch (error) {
+        console.error('Error fetching card details:', error);
+        setCardDetails(null);
       }
-      const data = await response.json();
-      setData(data.data);
     };
-    fetchCard();
+
+    if (id) {
+      fetchCardDetails(id)
+        .then(data => setCardDetails(data))
```

Comment:
You rewriting the setCardDetails here and we lost all information about the card on the page. 
<img width="922" alt="image" src="https://github.com/Code-the-Dream-School/ffprac-team6/assets/106414157/e3ee92d1-9786-47c6-a671-353e72e2b4d2">